doc/guide/writing-persistent-objects.rst

```diff
@@ -503,6 +512,121 @@ ghost:
     >>> book._p_changed, bool(book._p_oid)
     (None, True)
 
+Things you can do, but need to carefully consider (advanced)
+============================================================
+
+While you can do anything with a persistent subclass that you can with
+a normal subclass, certain things have additional implications for
+persistent objects. These often show up as performance issues.
+
+Implement ``__eq__`` and ``__hash__``
+-------------------------------------
+
+When you store an entry into a Python ``dict`` (or the persistent
+variant ``PersistentMapping``) the key's ``__eq__`` and
+``__hash__`` methods are used to determine where to store the value.
+Later they are used to look it up via ``in`` or ``__getitem__``.
+
+When that ``dict`` is later loaded from the database, the internal
+storage is rebuild from scratch. This means that every key has its
+``__hash__`` method called at least once, and may have its ``__eq__``
+method called many times.
+
+By default, every object, including persistent objects, inherits an
+implementation of ``__eq__`` and ``__hash__`` from :class:`object`.
+These default implementations are based on the object's *identity*,
+that is, its unique identifier within the current Python process.
+Calling, them, therefore is very fast, even on ghosts, and doesn't
```

Comment:
make one of the uses a "ghosts" uses a reference to the section on ghosts, using:

    :ref:`ghost <ghost-label>`